src/Servers/Kestrel/test/InMemory.FunctionalTests/Http3/WebTransportTests.cs

```diff
@@ -56,34 +58,43 @@ public async Task WebTransportHandshake_ClientToServerPasses()
         ((long)Http3ErrorCode.ProtocolError),
         nameof(HeaderNames.Method), "GET", // incorrect method (verifies that webtransport doesn't break regular Http/3 get)
         nameof(HeaderNames.Protocol), "webtransport",
-        nameof(HeaderNames.Scheme), "http",
+        nameof(HeaderNames.Scheme), "https",
         nameof(HeaderNames.Path), "/",
         nameof(HeaderNames.Authority), "server.example.com",
         nameof(HeaderNames.Origin), "server.example.com")]
     //[InlineData(
     //    ((long)Http3ErrorCode..ProtocolError,
     //    nameof(HeaderNames.Method), "CONNECT",
     //    nameof(HeaderNames.Protocol), "webtransport",
-    //    nameof(HeaderNames.Scheme), "http", // incorrect scheme <-- it should be https but the tests don't work with it so I commented it out (also it is given by http/3)
+    //    nameof(HeaderNames.Scheme), "http", // incorrect scheme <-- this is a given by http/3 so is there a need to test this?
     //    nameof(HeaderNames.Path), "/",
     //    nameof(HeaderNames.Authority), "server.example.com",
     //    nameof(HeaderNames.Origin), "server.example.com")]
+    [InlineData(
+        ((long)Http3ErrorCode.ProtocolError),
+        nameof(HeaderNames.Method), "CONNECT",
+        nameof(HeaderNames.Protocol), "NOT_webtransport",
```

Comment:
There's still no logic to reject a "NOT_webtransport" protocol. This test is only passing because the request is getting rejected due to a missing protocol.

We should use a unique `Http3StreamErrorException.Message` for each rejection reason and we should always be always using the `matchExpectedErrorMessage` to verify were rejecting the request for the right reasons.

https://github.com/dotnet/aspnetcore/blob/main/src/Servers/Kestrel/shared/test/Http3/Http3InMemory.cs#L576-L589

`matchExpectedErrorMessage?.Invoke(expectedErrorMessage);` is never going to fail though 🤣 It looks like we need to update `WaitForStreamErrorAsync` to check `matchExpectedErrorMessage` against the `abortReason.Message` passed into `TestStreamContext.Abort(ConnectionAbortedException abortReason)` while we're at it.

Reply:
Or maybe it can look at the `LogMessages` like `Http2TestBase` does.

https://github.com/dotnet/aspnetcore/blob/0d1a2a869b8d00e9ddb1eb8f7ce98e261dc1becd/src/Servers/Kestrel/test/InMemory.FunctionalTests/Http2/Http2TestBase.cs#L1287-L1290

Reply:
Ohh, ok thank you for clarifying. I didn't understand what you meant before. I will make sure to fix this!